src/components/Notifications/EventLog/usePrimaryToolbarFilterConfigWrapper.tsx

```diff
@@ -0,0 +1,180 @@
+                chips: (filters.application as string[])?.map(filterApplication => {
+                    const appDisplayName = bundle?.children?.find(application => application.name === filterApplication)?.displayName;
+                    return {
+                        name: appDisplayName as string,
+                        value: filterApplication,
+                        isRead: true
+                    };
+                }) || applicationChips
+            };
+        }) || [];
+
+        return eventLogCustomFilters;
+    }, []);
+
+    React.useEffect(() => {
+        setCustomFilters(mapToEventLogCustomFilter(filters, bundles.length !== 0 ? bundles : undefined));
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [ bundles, mapToEventLogCustomFilter ]);
+
+    toolbarConfig.filterConfig.items[1] = useMemo(() => {
+        return {
+            label: 'Application',
+            type: 'custom',
+            filterValues: {
+                children: <EventLogTreeFilter
+                    groups={ bundles }
+                    placeholder={ 'Filter by Application' }
+                    filters={ customFilters }
+                    updateFilters={ setCustomFilters }
+                />
+            }
+        } as any;
+    }, [ bundles, customFilters ]);
+
+    toolbarConfig.activeFiltersConfig.filters = React.useMemo(() => {
+        const activeFilters = toolbarConfig.activeFiltersConfig.filters;
+        const nonCustomFilters = activeFilters.filter(activeFilter => !!activeFilter && !(activeFilter as EventLogCustomFilter).bundleId);
+        return nonCustomFilters.concat(customFilters);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [ filters, customFilters ]);
+
+    const { activeBundles, activeApplications } = React.useMemo(() => {
+        // While bundles are empty, assume network request is still pending
+        if (bundles.length === 0) {
+            return { activeBundles: filters.bundle, activeApplications: filters.application };
+        }
+
+        const activeBundles: string[] = [];
+        const activeApplications: string[] = [];
+        customFilters.forEach(customFilter => {
+            activeBundles.push(customFilter.bundleId);
+
+            const bundle = bundles.find(bundle => bundle.name === customFilter.bundleId) as Schemas.Facet;
+            const chipValues = customFilter.chips?.map(chip => chip.value) as string[] | undefined;
+
+            if (chipValues && bundle.children?.some(application => !chipValues.includes(application.name))) {
+                chipValues.forEach(chipValue => {
+                    if (!activeApplications.includes(chipValue)) {
+                        activeApplications.push(chipValue);
+                    }
+                });
+            }
+        });
+
+        return { activeBundles, activeApplications };
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [ bundles, customFilters ]);
```

Comment:
Check missing dependencies, usually we don't need to skip this lint rule.

Reply:
Instead of doing it this way, which is using a memo to memoize the current filtered bundles and applications to keep the URL params up to date and having to omit dependencies... I will use `produce` function inside the `setFilters.bundle` and `setFilters.application`.

Reply:
I changed this design completely because of this comment, so this is no longer relevant, resolving....